src/zcash/IncrementalMerkleTree.hpp

```diff
@@ -289,52 +288,40 @@ class OrchardMerkleFrontier
     OrchardMerkleFrontier& operator=(const OrchardMerkleFrontier& frontier)
     {
         if (this != &frontier) {
-            inner.reset(orchard_merkle_frontier_clone(frontier.inner.get()));
+            inner = frontier.inner->box_clone();
         }
         return *this;
     }
 
     template<typename Stream>
     void Serialize(Stream& s) const {
-        RustStream rs(s);
-        if (!orchard_merkle_frontier_serialize(inner.get(), &rs, RustStream<Stream>::write_callback)) {
-            throw std::ios_base::failure("Failed to serialize v5 Orchard tree");
-        }
+        inner->serialize(s);
     }
 
     template<typename Stream>
     void Unserialize(Stream& s) {
-        RustStream rs(s);
-        OrchardMerkleFrontierPtr* tree = orchard_merkle_frontier_parse(
-                &rs, RustStream<Stream>::read_callback);
-        if (tree == nullptr) {
-            throw std::ios_base::failure("Failed to parse v5 Orchard tree");
```

Comment:
In PR gardening, @daira reminded me that we handle `std::ios_base::failure` specially in `ProcessMessages()`:
- We send a `reject` message.
- If the message is one of a few common cases, we log it quietly.
- If otherwise, we log it noisily.

The current change in this PR would just cause the exception to be logged noisily. While that was the case previously as well (this error message didn't match any of the common cases, in part because we couldn't expose the underlying C++ stream error), we now _can_ potentially do so due to `cxx` enabling us to pass error information from C++ to Rust and back to C++. So we should catch exceptions here and re-raise them as `std::ios_base::failure`.